### ISSUE TITLE
Make compute_mse_and_acc func on pg 354 more elegant

### ISSUE THREAD

**Issue by rasbt** [rasbt/machine-learning-book#55]
actually, I am not sure why I wasn't computing it with 
    loss = np.sum((onehot_targets - probas)**2)

(instead of mean) in the loop and then using 

    mse = mse/num_examples

instead of 

    mse = mse/i

Let me make a note to update this!

**Comment by rasbt** [comment]
actually, upon revisiting this, the original was better

**Comment by pavlo-yanchenko** [comment]
I've really found this `i` quite confusing, predominantly due to (in fact) `i = n_batches - 1` (because enumerate started with 0), and so `mse = mse / i` not equal to batch mses mean as `mse = mse / n_batches`. Of course, there is no significant difference even on 50 batches of X_val sets but it still confuses me...

**Comment by pavlo-yanchenko** [comment]
Let me propose another variant of the function without confusing `i`, which leads to obtaining slightly incorrect MSE (not because of "rounding error" as that is explained in the book, but because of the `i = n_batches - 1`

```
def compute_mse_and_acc(model, X, y, num_labels=10, minibatch_size=100):
     blosses, correct_pred = 0, 0
     n_batches = int(X.shape[0] / minibatch_size)
     n_samples = n_batches * minibatch_size
     mbatch_gen = minibatch_generator(X, y, minibatch_size)
     for features, targets in mbatch_gen:
        _, probas = model.forward(features)
        predicted_labels = np.argmax(probas, axis=1)
        one_hot_targets = int_to_onehot(targets, num_labels=num_labels)
        batch_loss = np.mean((one_hot_targets - probas)**2)
        correct_pred += (predicted_labels == targets).sum()
        blosses += batch_loss
     mse = blosses / n_batches
     acc = correct_pred/n_samples
     return mse, acc
```
